### PR TITLE
[MAINTENANCE] Remove v2 api expectations tests

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- We are removing the deprecated v2 api. This PR removes the expectations test runner.

### Definition of Done

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code

### CODE REVIEW

**Review by Shinnnyshinshin — APPROVED** [review]
🚀

File: tests/datasource/test_new_datasource.py [hunk 1]
```diff
@@ -236,31 +234,26 @@ def test_basic_pandas_datasource_v013_self_check(basic_pandas_datasource_v013):
     }
 
 
-def test_basic_spark_datasource_self_check(basic_spark_datasource):
+def test_basic_spark_datasource_self_check_spark_app_name(basic_spark_datasource):
```
Comment (alexsherstinsky) [inline tests/datasource/test_new_datasource.py]: @anthonyburdi Thinking of proposing that we do not change the test name (for  now).

Comment (anthonyburdi) [reply tests/datasource/test_new_datasource.py]: Agree! But I'm not changing the test name, this is an additional test. The `test_basic_spark_datasource_self_check` still exists below.

**Review by alexsherstinsky — APPROVED** [review]
LGTM -- thank you!

**Comment by anthonyburdi** [comment]
> LGTM -- thank you!

Thanks for the helpful conversation @alexsherstinsky !!

**Pull request merged**